sstables_loader.cc

```diff
@@ -242,11 +244,31 @@ future<> sstable_streamer::stream(std::function<void(unsigned)> on_streamed) {
     co_await stream_sstables(full_partition_range, std::move(_sstables), std::move(on_streamed));
 }
 
+bool tablet_sstable_streamer::tablet_in_scope(locator::tablet_id tid) const {
+    if (_stream_scope == stream_scope::all) {
+        return true;
+    }
+
+    const auto& topo = _erm->get_topology();
+    for (const auto& r : _tablet_map.get_tablet_info(tid).replicas) {
+        if (_stream_scope == stream_scope::node && topo.is_me(r.host)) {
+            return true;
+        }
+        if (_stream_scope == stream_scope::rack && topo.get_location(r.host) == topo.get_location()) {
+            return true;
+        }
+        if (_stream_scope == stream_scope::dc && topo.get_datacenter(r.host) == topo.get_datacenter()) {
+            return true;
+        }
+    }
```

Comment:
Note: it's better to use switch, since the compiler warns you if you add an enum member and don't extend the switch.